### PR TITLE
FOUNDATIONS: Remove Post Impression 

### LINKED ISSUE #242: FOUNDATIONS: Remove Post Impression

### PR DESCRIPTION
**Pull request by Javlonbek-dev**
Closes #242

### CODE REVIEW

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs [hunk 1]
```diff
@@ -40,6 +40,30 @@ private static void ValidatePostImpressionIsNotNull(PostImpression postImpressio
             }
         }
 
+        private void ValidatePostImpressionIds(Guid postId, Guid profileId)
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs:43]: Can you change this to ValidatePostImpressionOnRemove and make it work in the same way as ValidatePostImpressionOnAdd.

The way you have it here, it only evaluates 1 before you move on to the next one.
If you use the `Validate(...)` then you will have the error details of both in the validation summary

 Validate(
    (Rule: IsInvalid(postId), Parameter: nameof(postId)),
    (Rule: IsInvalid(profileId), Parameter: nameof(profileId)));

Comment (Javlonbek-dev) [reply Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs]: Done

File: Taarafo.Core/Services/Foundations/PostImpressions/IPostImpressionService.cs [hunk 2]
```diff
@@ -13,6 +14,6 @@ public interface IPostImpressionService
     {
         ValueTask<PostImpression> AddPostImpressions(PostImpression postImpression);
         IQueryable<PostImpression> RetrieveAllPostImpressions();
-
+        ValueTask<PostImpression> RemovePostImpressionByIdAsync(Guid postId, Guid profileId);
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/IPostImpressionService.cs:17]: Change to ValueTask<PostImpression> RemovePostImpressionAsync(PostImpression postImpression);

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs [hunk 3]
```diff
@@ -38,5 +39,18 @@ public partial class PostImpressionService : IPostImpressionService
 
         public IQueryable<PostImpression> RetrieveAllPostImpressions() =>
            TryCatch(() => this.storageBroker.SelectAllPostImpressions());
+
+        public ValueTask<PostImpression> RemovePostImpressionByIdAsync(Guid postId, Guid profileId) =>
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs:43]: Change to 
`public ValueTask<PostImpression> RemovePostImpressionAsync(PostImpression postImpression) =>`

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs [hunk 4]
```diff
@@ -38,5 +39,18 @@ public partial class PostImpressionService : IPostImpressionService
...
+            TryCatch(async () =>
+            {
+                ValidatePostImpressionOnRemove(postId, profileId);
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs:46]: Change to
`ValidatePostImpressionOnRemove(postImpression);`

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs [hunk 5]
```diff
@@ -40,6 +40,22 @@ private static void ValidatePostImpressionIsNotNull(PostImpression postImpressio
             }
         }
 
+        private static void ValidatePostImpressionOnRemove(Guid postId, Guid profileId)
+        {
+
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs:45]: Add:
`ValidatePostImpressionIsNotNull(postImpression);`

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Some small changes required

**Review by cjdutoit — APPROVED** [review]

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Please fix build issue

File: Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Validations.RemoveById.cs [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.PostImpressions
+{
+    public partial class PostImpressionServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidatonExceptionOnRemoveIfIdIsInvalidAndLogItAsync()
+        {
+            //given
+            PostImpression randomPostImpression = CreateRandomPostImpression();
```
Comment (cjdutoit) [inline Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Validations.RemoveById.cs:22]: Revisit this test.  You are creating a valid PostImpression, but you wish to test for invalid Id's

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Please revisit test to fix build issues

File: Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Validations.RemoveById.cs (hunk 6 above)
Comment (Javlonbek-dev) [reply Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Validations.RemoveById.cs]: Done

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**